OpenSim/Common/Component.h

```diff
@@ -712,6 +716,7 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     */
     template <class C = Component>
     C& updComponent(const std::string& name) {
+        clearObjectIsUpToDateWithProperties();
```

Comment:
Does this mean that the simple act of fetching a writable reference to a Component invalidates its up-to-dateness? Maybe that's okay—though users will need to be very careful to use `getComponent()` rather than `updComponent()`: there will be situations where both will work but `updComponent()` will take longer.